### PR TITLE
switch to amrex::ParallelFor

### PR DESCRIPTION
**Pull request by zingale**
this gives use better backtraces

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
all tests pass:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-05-07-006/index.html

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -2890,7 +2890,8 @@ Castro::normalize_species (MultiFab& S_new, int ng)
         // Ensure the species mass fractions are between small_x and 1,
         // then normalize them so that they sum to 1.
 
-        AMREX_PARALLEL_FOR_3D(bx, i, j, k,
+        amrex::ParallelFor(bx,
+        [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:2894]: Let's make these `AMREX_GPU_HOST_DEVICE`, please.

Comment (zingale) [reply Source/driver/Castro.cpp]: updated

**Comment by zingale** [comment]
I fixed the last AMREX_PARALLEL_FOR_3D in Riemann

File: Source/hydro/riemann.H [hunk 2]
```diff
@@ -54,7 +54,7 @@ Castro::pstar_bisection(Real& pstar_lo, Real& pstar_hi,
                         const Real gdot, const Real gmin, const Real gmax,
                         const int lcg_maxiter, const Real lcg_tol,
                         Real& pstar, Real& gamstar,
-                        bool& converged, GpuArray<Real, PSTAR_BISECT_FACTOR*HISTORY_SIZE>& pstar_hist_extra) {
+                        bool& converged, GpuArray<Real, PSTAR_BISECT_FACTOR*HISTORY_SIZE> pstar_hist_extra) {
```
Comment (maxpkatz) [inline Source/hydro/riemann.H:57]: Don't we want to keep this by reference since we're writing to `pstar_hist_extra`?

Comment (zingale) [reply Source/hydro/riemann.H]: changed

**Pull request merged**